doc/proposals/composable-integration-in sdk.md

```diff
@@ -0,0 +1,173 @@
+
+- \[x\] Enhancement is `implementable`
+- \[x\] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created in [operator-sdk/doc][operator-sdk-doc]
+
+
+## Summary
+
+Kubernetes object specifications often require constant values for their fields. When deploying an entire application
+with many different resources, this limitation often results in the need for staged deployments, because some resources have to be deployed first in order to determine what data to provide for the specifications of dependent resources. This undermines the declarative nature of Kubernetes object specification and requires workflows, manual step-by-step instructions and/or brittle automated scripts for the deployment of applications as a whole.
+
+The Composable SDK can be used to add cross-resource references to any existing CRD, so that values no longer
+need to be hardwired. This feature allows dynamic configuration of a resource, meaning that its fields can be 
+resolved after it has been deployed. 
+
+See this [tutorial](https://github.com/IBM/composable/blob/master/sdk/docs/tutorial.md), in which we add cross-references to the `memcached-operator` using the Composable SDK.
+
+The Composable SDK has been implemented and open-sourced at https://github.com/IBM/composable/tree/master/sdk.
+It is also related to the Composable Operator released on operatorhub.io:
+https://github.com/IBM/composable.
+
+
+## Motivation
+
+To add support for cross-resource references in CRD definitions and controllers (Composable SDK) and make this
+functionality available through the Operator-SDK.
+
+## Goals
+
+- Provide cross-resource reference types, which can be used in CRD type definitions.
+- Provide resolution functions, which can be called from within a reconciler to resolve cross-resource references.
+
+
+## Proposal
+
+### User Story 1
+
+As an operator developer, I wish not to have the value of a field be hard-wired. These are typically
+values that are computed dynamically when an application is deployed. For example, the admin URL for a Kafka
+deployment is known only after Kafka has been successfully deployed. If my CRD requires the URL, then
+rather that requiring a string value for that field, I can define it to be a reference to another object
+(perhaps the data is in the status field of the Kubernetes object managing Kafka).
+
+### User Story 2
+
+Often the yamls for a collection of resources require the same data in many places (e.g., a host name or port number).
+Rather than hardwiring the same value everywhere, I can define cross-resource references to a unique configmap that contains 
+the data. This makes it easier to change the data if needed, and can allow all resources to be configured
+dynamically (i.e. changes in the configmap are absorbed and resource that point to it are updated).
+
```

Comment:
The same need could be solved by just GET object in the controller and then, updating the CR resource with the values. Not necessary is required to add all values used in the CR. It can be obtained programmatically, dynamically and in realtime as well.  

IHMO the users stories needs to be improved because the idea here would not be allowed something that is not possible today. It is just to add a facility, I mean another way to get RefFrom without need to implement it.

Reply:
The idea is that rather than hardwiring that in the controller, each CR of the CRD could have fields that point to different objects. So this gives a lot of flexibility.